cloud-spanner/modules/cloudspanner/variables.tf

```diff
@@ -51,7 +51,7 @@ variable "ddl_queries" {
     default = [
       "CREATE TABLE users ( userId STRING(36) NOT NULL, businessEmail STRING(50), fullName STRING(36), password STRING(100), photoUrl STRING(250), provider STRING(20),) PRIMARY KEY(userId)",
       "CREATE TABLE companies (companyId STRING(36) NOT NULL, companyName STRING(30), companyShortCode STRING(15), created_at TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp = true), ) PRIMARY KEY(companyId)",
-      "CREATE TABLE companyStocks (companyStockId STRING(36) NOT NULL, companyId STRING(36) NOT NULL, companyShortCode STRING(10), exchangeName STRING(50), exchangeMic STRING(50), timezone STRING(50), shares FLOAT64, requestId STRING(15), open FLOAT64, volume FLOAT64, currentValue FLOAT64, date FLOAT64, close FLOAT64, dayHigh FLOAT64, dayLow FLOAT64, adjHigh FLOAT64, adjLow FLOAT64, adjClose FLOAT64, adjOpen FLOAT64, adjVolume FLOAT64, timestamp TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp = true), ) PRIMARY KEY(companyStockId)",
+      "CREATE TABLE companyStocks (companyStockId STRING(36) NOT NULL, companyId STRING(36) NOT NULL, companyShortCode STRING(36), open NUMERIC, volume NUMERIC, currentValue NUMERIC, date FLOAT64, close NUMERIC, dayHigh NUMERIC, dayLow NUMERIC, timestamp TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp=true)) PRIMARY KEY(companyStockId)",
       "CREATE TABLE simulations (sId STRING(36) NOT NULL, companyId STRING(36) NOT NULL, status BOOL, createdAt TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp = true),) PRIMARY KEY(sId)",
```

Comment:
See comment below on `deleteCompany` function: I think it would be good to define `FOREIGN KEY` constraints on `companyStocks.companyId` and `simulations.companyId`.

Reply:
the above issue has been resolved.

Reply:
The DDL has been updated.
[https://github.com/searceinc/spanner-terraform-example/blob/manage-companies/modules/cloud-spanner/variables.tf](url)

Reply:
The above URL returns a 404 for me. Are you sure it's the correct one?

Reply:
Regarding those changes:

https://github.com/searceinc/spanner-terraform-example/blob/10fea4ddacc895de450dd5c742d6e3bfe4e1dbad/modules/cloud-spanner/variables.tf#L57

The above unique index definition will require that all __combinations__ of (companyName, companyShortcode) are unique. That means that there may be several companies with the same name and several companies with the same code. The application searches for companies by selecting `WHERE companyName=@name OR companyShortcode=@code`. That means that each column individually should be unique, which would mean two separate unique indexes.

Reply:
https://github.com/searceinc/spanner-terraform-example/blob/10fea4ddacc895de450dd5c742d6e3bfe4e1dbad/modules/cloud-spanner/variables.tf#L58-L59

These tables are created as child tables (INTERLEAVED IN PARENT) of `companies`, and __also__ define a foreign key constraint to the table `companies`. It should only be one or the other.

- Use `INTERLEAVED IN PARENT` when the rows of the parent and child tables should be physically stored together. I don't think that is appropriate in this case, considering the fact that there will probably be a very large number of child records for each parent record.
- Use `FOREIGN KEY` when the database should enforce that a child row must use a valid parent id, and that the parent record may only be deleted once all the child records have been deleted. Foreign keys does not instruct the database to store the parent and child rows physically together. I think this option is the best suited for this specific data model.